executor/src/lib.rs

```diff
@@ -153,15 +166,54 @@ impl Executor {
 
         // Spawn the executor's core.
         let executor_handle = Core::<State>::spawn(
-            store,
+            batch_store,
             execution_state,
             tx_reconfigure.subscribe(),
             /* rx_subscriber */ rx_executor,
+            restored_consensus_output,
         );
 
         // Return the handle.
         info!("Consensus subscriber successfully started");
 
         Ok(vec![subscriber_handle, executor_handle])
     }
+
+    pub async fn get_restored_consensus_output<State>(
+        consensus_store: Arc<ConsensusStore>,
+        certificate_store: Store<CertificateDigest, Certificate>,
+        execution_state: Arc<State>,
+    ) -> Vec<ConsensusOutput>
+    where
+        State: BatchExecutionState + Send + Sync + 'static,
+        State::Error: Debug,
+    {
+        let mut restored_consensus_output = Vec::new();
+        let consensus_max_seq = consensus_store
+            .read_last_consensus_index()
+            .expect("Failed to load consensus index from store");
+
```

Comment:
looking on the semantics of the `read_last_consensus_index` and how that is being stored/used I believe it would be better to rename the variable to something like `consensus_next_index` , as practically this is what is being stored in there. Looking in the consensus code it seems we assign the current `consensus_index` to a sequenced certificate, we increment its value +1 , and then we store the index in store. So in the end, when you read the `read_last_consensus_index` is the "next" index that will be assigned to the next certificate that will be sequenced.